### PR TITLE
W-11091826 Manage Expenditures Component

### PR DESCRIPTION
**Pull request by mldyang**
GUS: [W-11091826: Outbound Funds/NPSP Extension - Manage Expenditures component](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00000wG44tYAC/view)

# Critical Changes

# Changes
Enabling Lightning Web Security surfaced an issue with the Manage Expenditure Component where Disbursement record page does not load. This PR fixes the issue.

# Issues Closed

# New Metadata

# Deleted Metadata

# Definition of Done

Refer to [Definition of Done](https://salesforce.quip.com/9P7hAOPHJJyU) to see any additional details for the items below:

- [x] Any net new LWC work has JEST test coverage 50% or above
- [x] Default Sa11y tests pass for all LWC components
- [x] 🔒 Secure both Front-end (LWC) & back-end (Apex) as necessary
- [x] 🔑 Grant users access in Permission Sets (Object, Field, Apex Class) as necessary
- [x] Link the pull request and work item by PR comment and Chatter post respectively, e.g. GUS: [W-0000000: Work Name]()
- [x] Make sure that ACs are updated (if any gaps)
- [ ] **All acceptance criteria have been met**
  - [x] Developer
  - [ ] Code Reviewer
- [x] Pull Request contains draft release notes
- [x] Labels, help text, and customer facing messages are reviewed by Docs
- [ ] QE story level testing completed

### CODE REVIEW

File: robot/OutboundFundsNPSP/tests/create_contact.robot [hunk 1]
```diff
@@ -41,5 +41,5 @@ Validate Contact
     Page Should Contain  ${first_name} ${last_name}
     # Validate via API
     &{contact} =     Salesforce Get  Contact  ${contact_id}
-    Should Be Equal  ${first_name}  &{contact}[FirstName]
-    Should Be Equal  ${last_name}  &{contact}[LastName]
+    Should Be Equal  ${first_name}  ${contact['FirstName']}
+    Should Be Equal  ${last_name}  ${contact['LastName']}
```
Comment (mldyang) [inline robot/OutboundFundsNPSP/tests/create_contact.robot:45]: This change fixes the error in Tests/Create Contact - `Value of variable '&{contact}[FirstName]' is not dictionary or dictionary-like.`

File: robot/OutboundFundsNPSP/tests/browser/Requirements/Requirements.robot [hunk 2]
```diff
@@ -45,5 +45,6 @@ Add a Requirement on a Funding Request
     Click Save
     wait until modal is closed
     Click Related List Link With Text           ${req_name}
+    Select Tab                                  Details
```
Comment (mldyang) [inline robot/OutboundFundsNPSP/tests/browser/Requirements/Requirements.robot:48]: This change fixes the error in Tests/Browser/Requirements - `locator not found` by actually navigating to the Details tab (default landing tab is Related)

File: robot/OutboundFundsNPSP/resources/locators_54.py [hunk 3]
```diff
@@ -39,6 +39,10 @@
         "verify_header": "//div[contains(@class, 'entityNameTitle')]",
         "verify_details": "//div[contains(@class, 'slds-form-element')][.//span[text()='{}']]//following-sibling::div[.//span[contains(@class, 'test-id__field-value')]]/span",
     },
+    "tabs":{
```
Comment (mldyang) [inline robot/OutboundFundsNPSP/resources/locators_54.py:42]: This change is in support of "Select Tab," for Tests/Browser/Requirements fix

File: robot/OutboundFundsNPSP/resources/OutboundFundsNPSP.py [hunk 4]
```diff
@@ -240,6 +241,30 @@ def verify_row_count(self, value):
             value, count
         )
 
+    @capture_screenshot_on_error
+    def select_tab(self, title):
```
Comment (mldyang) [inline robot/OutboundFundsNPSP/resources/OutboundFundsNPSP.py:245]: This change is in support of "Select Tab," for Tests/Browser/Requirements fix

File: robot/OutboundFundsNPSP/doc/Keywords.html [hunk 5]
```diff
@@ -285,6 +285,18 @@ <h2 title="robot/OutboundFundsNPSP/resources/OutboundFundsNPSP.py">
                                 </td>
                             </tr>
 
+                            <tr class="kwrow" id="OutboundFundsNPSP.py.Select Tab">
```
Comment (mldyang) [inline robot/OutboundFundsNPSP/doc/Keywords.html:288]: This change is in support of "Select Tab," for Tests/Browser/Requirements fix

File: robot/OutboundFundsNPSP/resources/OutboundFundsNPSP.py [hunk 6]
```diff
@@ -2,16 +2,17 @@
 import random
 import string
 import warnings
+import time
 
 from BaseObjects import BaseOutboundFundsNPSPPage
 from robot.libraries.BuiltIn import RobotNotRunningError
-from locators_52 import outboundfundsnpsp_lex_locators as locators_52
+from locators_54 import outboundfundsnpsp_lex_locators as locators_54
```
Comment (mldyang) [inline robot/OutboundFundsNPSP/resources/OutboundFundsNPSP.py:9]: Updating from api 52 to 54 to fix an error encountered while running Test/Browser/GAUExpenditurePermissions.robot test

File: robot/OutboundFundsNPSP/resources/locators_54.py [hunk 7]
```diff
@@ -13,7 +13,7 @@
         "edit_title": "//h2[contains(@class, 'title') and text()='{}']",
         "list": "//div[contains(@class,'forcePageBlockSectionRow')]/div[contains(@class,'forcePageBlockItem')]/div[contains(@class,'slds-hint-parent')]/div[@class='slds-form-element__control']/div[.//span[text()='{}']][//div[contains(@class,'uiMenu')]//a[@class='select']]",
         "text_field": "//div[contains(@class, 'uiInput')][.//label[contains(@class, 'uiLabel')][.//span[text()='{}']]]//*[self::input or self::textarea]",
-        "dropdown_field": "//lightning-combobox[./label[text()='{}']]/div//input[contains(@class,'combobox__input')]",
+        "dropdown_field": "//lightning-combobox[./label[text()='{}']]/div//button[contains(@class,'slds-combobox__input')]",
```
Comment (mldyang) [inline robot/OutboundFundsNPSP/resources/locators_54.py:16]: This change fixes the error in Tests/Browser/GAUExpenditurePermissions - `Element with locator '//lightning-combobox[./label[text()='Status']]/div//input[contains(@class,'combobox__input')]' not found.`

**Review by davidmreed — APPROVED** [review]

**Comment by davidmreed** [comment]
Approved but recommend running `black` to clean up the lint.

**Comment by mldyang** [comment]
GUS: [W-11091826: Outbound Funds/NPSP Extension - Manage Expenditures component
](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00000wG44tYAC/view)

File: robot/OutboundFundsNPSP/tests/browser/Requirements/Requirements.robot (hunk 2 above)
Comment (spelak-salesforce) [reply robot/OutboundFundsNPSP/tests/browser/Requirements/Requirements.robot:48]: 🎉 !!

File: robot/OutboundFundsNPSP/tests/create_contact.robot (hunk 1 above)
Comment (spelak-salesforce) [reply robot/OutboundFundsNPSP/tests/create_contact.robot:45]: Thank you@

**Review by spelak-salesforce — APPROVED** [review]
All code changes look great + all tests pass.  Great work @mldyang !

**Pull request merged**